tuner/tuner/candidate_gen.py

```diff
@@ -196,6 +246,25 @@ def generate_configs_and_td_specs(
             break
         tune_logger.debug(f"Solution #{i+1}: {config}")
         td_spec_module = dispatch_tuner.get_td_spec(input_module, config)
+        current_matchers = get_matcher_names_from_td_spec(td_spec_module)
+
+        should_link, new_warned_matchers = check_td_spec_matchers_overlap(
+            starter_matchers, current_matchers, warned_overlap_matchers
+        )
```

Comment:
If you make `check_td_spec_matchers_overlap` accept td specs as arguments and call `get_matcher_names_from_td_spec` internally, you can make it take a list of tuning specs. For the case with no starter spec, the list will always have a single element.